src/analyses/constant_propagator.cpp

```diff
@@ -339,19 +318,46 @@ bool constant_propagator_domaint::valuest::is_constant_address_of(
 /// Do not call this when iterating over replace_const.expr_map!
 bool constant_propagator_domaint::valuest::set_to_top(const irep_idt &id)
 {
-  bool result = false;
+  replace_symbolt::expr_mapt::size_type n_erased=
+    replace_const.expr_map.erase(id);
+
+  INVARIANT(n_erased==0 || !is_bottom, "bottom should have no elements at all");
+
+  return n_erased>0;
+}
+
+/*******************************************************************\
 
-  replace_symbolt::expr_mapt::iterator r_it =
-    replace_const.expr_map.find(id);
+Function: constant_propagator_domaint::valuest::set_dirty_to_top
 
-  if(r_it != replace_const.expr_map.end())
+  Inputs:
+
+ Outputs:
+
+ Purpose:
+
+\*******************************************************************/
```

Comment:
Remove this non-doxygen  comment header

Reply:
AAAAARRRRGGGGG I thought I'd got all of them!

Fixed.